### PR TITLE
Dont include tags and categories unless they are present

### PR DESCRIPTION
**Pull request by ejaypcanaria**

### CODE REVIEW

File: lib/woocommerce_api/resources/product.rb [hunk 1]
```diff
@@ -22,8 +22,8 @@ def as_json(options=nil)
       end
 
       # Do not override categories and tags if an empty array is passed
-      wc_attributes['product'].delete('categories')
-      wc_attributes['product'].delete('tags')
+      wc_attributes['product'].delete('categories') unless wc_attributes['product']['categories'].present?
```
Comment (bradleypriest) [inline lib/woocommerce_api/resources/product.rb]: `.length/`.any?` maybe. I always get confused on what `.present?` means for an array

Comment (bradleypriest) [inline lib/woocommerce_api/resources/product.rb]: Oh man too much JS.  Length wouldn't even work.

Comment (ejaypcanaria) [inline lib/woocommerce_api/resources/product.rb]: :+1: `any?` makes more sense as it removes the `nil` in the array as well: `[nil].any? # => false`

**Comment by pmdeazeta** [comment]
```
categories  

List of product categories names (string). In write-mode need to pass a array of categories IDs (integer) (uses wp_set_object_terms())
```

```
tags	

List of product tags names (string). In write-mode need to pass a array of tags IDs (integer) (uses wp_set_object_terms())
```

**Comment by ejaypcanaria** [comment]
Ohh, they don't support string array for tags and categories :cry:

**Pull request closed without merging**